samza-kafka/src/main/scala/org/apache/samza/checkpoint/kafka/KafkaCheckpointManager.scala

```diff
@@ -157,7 +159,13 @@ class KafkaCheckpointManager(checkpointSpec: KafkaStreamSpec,
       },
 
       (exception, loop) => {
-        warn(s"Retrying failed checkpoint write to key: $key, checkpoint: $checkpoint for task: $taskName", exception)
+        if (loop.sleepCount >= MAX_RETRIES_ON_FAILURE) {
+          error(s"Exhausted failure retries: $MAX_RETRIES_ON_FAILURE when writing checkpoint: $checkpoint for task: $taskName.")
+          loop.done
```

Comment:
Do you need both done and throw?

"Exhausted $MAX_RETRIES retries when ..."

Reply:
Updated the patch in accordance to above comment.